packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -2351,6 +2351,54 @@ describe('TransactionController Integration', () => {
     });
   });
 
+  describe('when changing rpcUrl of networkClient', () => {
+    it('should update the trackingMap', async () => {
+      mockNetwork({
+        networkClientConfiguration: mainnetNetworkClientConfiguration,
+        mocks: [
+          // NetworkController
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x2',
+            },
+          },
+        ],
+      });
+      const { networkController, transactionController } =
+        await newController();
+      const startTrackinSpy = jest.spyOn(
+        transactionController,
+        'startTrackingByNetworkClientId',
+      );
+
+      await networkController.upsertNetworkConfiguration(
+        {
+          ...networkClientConfiguration,
+          rpcUrl: 'https://mock.rpc.url',
+        },
+        { setActive: false, referrer: 'https://mock.referrer', source: 'dapp' },
+      );
+
+      expect(startTrackinSpy).toHaveBeenCalledTimes(1);
```

Comment:
because changing the rpcUrl actually makes a new networkClient, i cant check the `networkClientId` because its a uuid